tests/FakeItEasy.IntegrationTests/FakingClassesTests.cs

```diff
@@ -12,10 +11,13 @@ public class FakingClassesTests
         [Test]
         public void Should_be_able_to_get_a_fake_value_of_uri_type()
         {
-            using (Fake.CreateScope(new NullFakeObjectContainer()))
-            {
-                A.Fake<Uri>();
-            }
+            // Arrange
+
+            // Act
+            var fake = A.Fake<Uri>();
+
+            // Assert
+            Fake.GetFakeManager(fake).Should().NotBeNull("because we should be able to create a fake Uri");

```

Comment:
> Is there a possible code path for `A.Fake<T>()` to return `null`?

Not that I can see, no. It'll make a fake or throw.

> we should really change this test to be `Should_be_able_to_get_a_fake_value_for_a_type_with_a_constructor_taking_arguments_of_that_type`

I've no objection to this. I think it's outside of the scope of this change, since the argument could be made whether we'd kept the container injection or not, but I'm happy to submit a PR.

> and the assertion should revert to asserting that no exception is thrown

I am not sure why the weaker "no exception thrown" is preferable to the stronger "we got an object back that is actually a Fake", since that's what we'd ultimately want.
Suppose a `null` were returned? That's undesirable. But just checking that no exception was thrown would not detect the problem.